dynamiq/nodes/agents/react.py

```diff
@@ -363,9 +363,9 @@ def _run_agent(self, config: RunnableConfig | None = None, **kwargs) -> str:
                         action, action_input = self._parse_action(llm_generated_output)
 
                     case InferenceMode.FUNCTION_CALLING:
-                        action = llm_result.output["tool_calls"][0]["function"]["name"].strip()
+                        action = llm_result.output["tool_calls"].values()[0]["function"]["name"].strip()
```

Comment:
Please double-check if this works fine. It seems this could fail and you need to wrap `values()` to list:
`list(llm_result.output["tool_calls"].values())[0]["function"]["name"].strip()`